src/coreclr/tools/dotnet-pgo/SPGO/LbrEntry.cs

```diff
@@ -74,7 +74,7 @@ public unsafe struct LbrTraceEventData64
         public Span<LbrEntry64> Entries(int totalSize)
         {
             IntPtr entriesOffset = Unsafe.ByteOffset(ref Unsafe.As<LbrTraceEventData64, byte>(ref this), ref Unsafe.As<LbrEntry64, byte>(ref _entries));
-            return MemoryMarshal.CreateSpan(ref _entries, (totalSize - (int)entriesOffset) / sizeof(LbrEntry64));
+            return MemoryMarshal.CreateSpan(ref Unsafe.AsRef<LbrEntry64>(Unsafe.AsPointer(ref _entries)), (totalSize - (int)entriesOffset) / sizeof(LbrEntry64));
```

Comment:
Won't that cause a GC hole? Same with above.

Reply:
Good catch. We either need to make this a `ref struct` with a big "DO NOT CLEAN UP THE REF PART" comment, or use `fixed` to grab the pointer instead.

Reply:
Yes it will. I'll push a fix.

Reply:
> We either need to make this a ref struct with a big "DO NOT CLEAN UP THE REF PART" comment, or use fixed to grab the pointer instead.

I think we can make these static functions that take the owning struct by ref.

Reply:
Can't we make it

```suggestion
            return MemoryMarshal.CreateSpan(ref _entries, (totalSize - (int)entriesOffset) / sizeof(LbrEntry64));
```

and add `[UnscopedRef]` to the method?

Reply:
Yes we can, that's an even smaller changeset, I'll do that. Thanks for the suggestion.

Reply:
This isn't a GC hole in this case. It is actually correct.

Reply:
We were trying to keep this narrow due to all the changes in flight. The problem here is the project is referencing reference assemblies from Preview 7 to build and the annotations here are simply wrong. Once the repo moves to RC1, this can be removed and updated appropriately.

Reply:
i shouldn't say "correct", rather it is "safe.

Reply:
I really wish this hadn't been touched.

Reply:
@AaronRobinsonMSFT I own this code -- I would prefer to have the code be correct by construction. I agree that it was fine before, but it was leaving an implicit assumption on how I am going to work with this code in the future.

Reply:
I see. This was something I was shepherding through for `ref` field support so it was in a state I knew to track. I will tag you if anything more comes up.

Reply:
Ah okay, sorry about that, I did not mean to step on your toes here. 

> I will tag you if anything more comes up.

Thanks, I appreciate it.

Reply:
Since I've also got the owner here, can we get this moved to build against the live build? Building against the repo's SDK is the root issue here.

Reply:
I can look into it, I'm not totally sure on the constraints about that (we are publishing dotnet-pgo as a nuget package, would that be affected?). I probably need to involve @davidwrighton here.
I assume it can be done separately?

Reply:
> I assume it can be done separately?

Absolutely. My only concern is this project is the only one I found that seems to use the repo's SDK and could be heavily impacted by complex language changes that force updates to the public API surface. This is a longer-term concern and just something to keep in mind as the language evolves to have more of these low-level friendly features.

Reply:
> My only concern is this project is the only one I found that seems to use the repo's SDK and could be heavily impacted by complex language changes that force updates to the public API surface

Everything under src/coreclr/tools builds against the repo's SDK. Building against live bits has build sequencing issues where one wouldn't be able to build crossgen2 without building libs first, or run crossgen2 without building the host.

If we want to switch to live, I would want to know what the VS debug experience for live built things looks like. I would still prefer we don't do that because I kind of like I don't need to build coreclr VM, host, or other stuff for NativeAOT development.

Reply:
> Building against live bits has build sequencing issues

Ah. Thanks @MichalStrehovsky. This is an important consideration to note. Perhaps this wouldn't justify the resulting dev-loop pain.